### DISCUSSION TITLE
NEED HELP: How to ensure playback of a complete playlist at specific time?

### DISCUSSION THREAD

**Discussion by matteo-daddi** [savonet/liquidsoap#1766] [Q&A]
[also sent to the savonet-users mailing list]

Hi everyone,

I need to playback a complete playlist (from start to end, not just a single or some media!) every day at the same time.
That playlist should "break" normal radio stream because it has higher priority.
I don't know the media's durations so I even don't know the complete playlist duration.

My "start time condition" is something like `20h00m00s`.

What I would like to do is something like this (NOT WORKING as desired):
```
radio = [...]
pls = playlist([...])
fallback(transition_length=0.,track_sensitive=false,[at({20h00m00s},pls), radio])
```
Here what's not working:
- The start time is not respected (there is a delay of around 20 seconds so the playlist starts more or less at 20h00m20s);
- The playlist is not played back entirely.

Don't know if it will work but things like `playlist.merge` are not an option because after this part of the script I need to maintain track boundaries to do other processing.

I've searched the mailing list archives and read the doc but I haven't found how to achieve this goal. Maybe it could be useful also to others (or maybe I'm missing something! :-D )

For the sake of completeness, I'm using liquidsoap 1.4.3+git@... I know that is not yet released but I think the question is version independent.

Thanks in advance!

**Comment by smimram** [comment]
I would do something like this:
```
radio = ...
pls = playlist(loop=false, "...")
s = fallback(track_sensitive=false, [pls, radio])
thread.when({20h00m00s}, {pls.reload()})
```
where we use `thread.when` to reload the contents of the playlist at the desired time.
